Yoast/wordpress-seo
# Publish flow

## Context


* Adds Yoast SEO to every step of the publication flow:
* While editing the document, the readability and SEO score are now visible in the document sidebar with a button that opens the Yoast sidebar.
![Screenshot 2020-06-04 at 23 07 29](https://user-images.githubusercontent.com/1488816/83811727-a7595980-a6ba-11ea-8803-6d9e13a4753b.png)
* When hitting publish, the readability and SEO score are added to the publish checks. When everything is green it functions as reassurance, otherwise there's a call to action to further optimize the content.
<img width="405" alt="Screenshot 2020-06-05 at 16 55 14" src="https://user-images.githubusercontent.com/1488816/83891391-5f374700-a74d-11ea-8fb3-f766edd1aead.png">
* After hitting publish, social sharing buttons are added to easily share the post on Facebook and Twitter.
<img width="363" alt="Screenshot 2020-06-05 at 16 55 26" src="https://user-images.githubusercontent.com/1488816/83891402-64949180-a74d-11ea-92fe-2d1109f5b4d4.png">

## Summary


This PR can be summarized in the following changelog entry:

* Adds SEO insights to every step of the publishing flow:
  * Adds the readability and SEO score to the document sidebar with the possibility to quickly open the Yoast sidebar.
  *  Adds the readability and SEO score to the publish checks. Gives reassurance or a call to action to further optimize the content.
  * Adds social sharing buttons to the post publish actions to enable easy sharing on Facebook and Twitter.

## Relevant technical choices:

* With the Pre publish and post publish panel I went with the standard fills Gutenberg offers for this.
* With the document sidebar I implemented a custom fill because the standard fill doesn't allow the panel to be initially opened. I did use the standard panel from Gutenberg.

## Test instructions

This PR can be tested by following these steps:

* Start a new post.
* Notice the Yoast SEO panel in the document sidebar.
* Click the button. (should only show if you have red and/or orange bullets)
* Notice the Yoast Sidebar opens.
* Hit Publish once (you need to have pre-publish checks enabled)
* Notice the Yoast SEO panel in the pre-publish sidebar.
* Click the button. (should only show if you have red and/or orange bullets)
* Notice the Yoast Sidebar opens.
* Finally publish the post.
* Notice the social buttons are added to the post publish checks.

## UI changes

## Documentation

## Quality assurance


Fixes #14465 
Fixes #14185
